scripts/generate/templates/docs/usage/configuration/sinks/README.md.erb

```diff
@@ -15,3 +15,52 @@ Sinks are last in the [pipeline][docs.pipelines], responsible for sending
 <%= components_table(metadata.sinks.to_h.values.sort) %>
 
 [+ request a new sink][url.new_sink]
+
+## Healthchecks
+
+Sinks may implement a healthcheck as a means for validating their configuration
+against the envionment and external systems. Ideally, this allows the system to
+inform users of problems such as insufficient credentials, unreachable
+endpoints, non-existant tables, etc. They're not perfect, however, since it's
+impossible to exhaustively check for issues that may happen at runtime.
+
+### Guidelines for writing healthchecks
+
+When implementing healthchecks, we prefer false positives to false negatives.
+This means we would prefer that a healthcheck pass and the sink then fail than
+to have the healthcheck fail when the sink would have been able to run
+successfully.
+
+A common cause of false negatives in healthchecks is performing an operation
+that the sink itself does not need. For example, listing all of the available S3
+buckets and checking that the configured bucket is in that list. The S3 sink
+doesn't need the ability to list all buckets, and a user that knows that may not
+have given it permission to do so. In that case, the healthcheck will fail due
+to bad credentials even through its credentials are sufficient for normal
+operation.
+
+This leads to a general strategy of mimicking what the sink itself does.
+Unfortunately, the fact that healthchecks don't have real events available to
+them leads to some limitations here. The most obvious example of this is with
+sinks where the exact target of a write depends on the value of some field in
+the event (e.g. an interpolated Kinesis stream name). It also pops up for sinks
```

Comment:
What would you propose for these circumstances then? Disable the health check?

Reply:
Yeah, if we can't know where data is actually going to be sent I'm not sure what we can do that isn't a guess. I'll add some clarification that in cases like this it's ok to fall back to doing nothing.